README.md

```diff
@@ -1,35 +1,466 @@
+
+If you would like to process the custom action completely on your own, you must add a custom listener to the Swrve SDK before its initialization. There is no default action for deeplinks in the Unity SDK.
+
+```
+private class CustomButtonListener : ISwrveCustomButtonListener {
+   public void OnAction (string customAction) {
+      // Custom button logic
+   }
+}
+SwrveComponent.Instance.SDK.GlobalCustomButtonListener = new CustomButtonListener();
+```
+
+iOS Push Notifications
+-
+
+1. Enable push notifications in the SwrveConfig object:
+
+  ```
+  #if UNITY_IPHONE
+  SwrveComponent.Instance.Config.PushNotificationEnabled = true;
+  #endif
+  ```
+
+2. Time the push permission request.
+
+  To request permission upon the triggering of certain events, include the following code:
+
+  Changes to `SwrveComponent.Instance.Config` must occur before calling `SwrveComponent.Instance.Init`.
+
+  ```
+  using System.Collections.Generic;
+
+  HashSet<string> pushNotificationEvents = new HashSet<string>();
+  pushNotificationEvents.Add("custom_event");
+
+  SwrveComponent.Instance.Config.PushNotificationEvents = pushNotificationEvents;
+  ```
+
+  To manage the exact timing for requesting permission to send push notifications, use the following code:
+
+  ```
+  SwrveComponent.Instance.SDK.NamedEvent("Swrve.push_notification_permission");
+  ```
+
+3. If you want to perform custom processing of the push notification payload, create the following class and add this code to the SDK initialization:
+
+  ```
+  using Swrve;
+
+  public class CustomPushNotificationListener : ISwrvePushNotificationListener {
+    public void OnRemoteNotification(RemoteNotification notification) {
+      // CUSTOM CODE HERE
+    }
+  }
+  SwrveComponent.Instance.SDK.PushNotificationListener = new CustomPushNotificationListener();
+  ```
+
+### Creating and uploading your iOS Cert ###
+
+To enable your app to send push notifications to Unity iOS devices, you require a push certificate. A push certificate authorizes an app to receive push notifications and authorizes a service to send push notifications to an app. For more information, see [How Do I Manage iOS Push Certificates?](http://docs.swrve.com/faqs/push-notifications/manage-ios-push-certificates-for-push-notifications/)
+
+### Provisioning your app ###
+
+Each time your app is published for distribution, you must provision it against the push certificate that has been created for the app.
+
+Android Push Notifications
+-
+
+Unity Android uses a native Android plugin (provided as part of the Unity SDK), so you must include this plugin in your project as detailed below.
+
+To integrate Swrve’s push notification functionality for Unity Android, you must be using Unity SDK version 2.9 or later and Unity Android Pro 4.0.0+.
```

Comment:
Not sure we need the mentions of SDK versions anymore as this will be tied in to a specific version